rlclientlib/serialization/payload_serializer.h

```diff
@@ -64,69 +63,68 @@ namespace reinforcement_learning {
         fbb.Finish(fb);
         return fbb.Release();
       }
-    };
 
-    struct outcome_single_serializer : payload_serializer<generic_event::payload_type_t::PayloadType_OutcomeSingle> {
-      static generic_event::payload_buffer_t numeric_event(float outcome) {
+      static generic_event::payload_buffer_t string_event(const char* outcome) {
         flatbuffers::FlatBufferBuilder fbb;
-        const auto evt = v2::CreateNumericEventSingle(fbb, outcome).Union();
-        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_NumericEventSingle, evt);
+        const auto evt = fbb.CreateString(outcome).Union();
+        auto fb = v2::CreateOutcomeEvent(fbb, v2::OutcomeValue_literal, evt);
         fbb.Finish(fb);
         return fbb.Release();
       }
 
-      static generic_event::payload_buffer_t string_event(const char* outcome) {
+      static generic_event::payload_buffer_t numeric_event(int index, float outcome) {
         flatbuffers::FlatBufferBuilder fbb;
-        const auto evt = v2::CreateStringEventSingleDirect(fbb, outcome).Union();
-        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_StringEventSingle, evt);
+        const auto evt = v2::CreateNumericOutcome(fbb, outcome).Union();
+        const auto idx = v2::CreateNumericIndex(fbb, index).Union();
+        auto fb = v2::CreateOutcomeEvent(fbb, v2::OutcomeValue_numeric, evt, v2::IndexValue_numeric, idx);
         fbb.Finish(fb);
         return fbb.Release();
       }
 
-      static generic_event::payload_buffer_t numeric_event(int index, float outcome) {
+      static generic_event::payload_buffer_t numeric_event(const char* index, float outcome) {
         flatbuffers::FlatBufferBuilder fbb;
-        const auto evt = v2::CreateNumericEventIndexed(fbb, outcome, index).Union();
-        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_NumericEventIndexed, evt);
+        const auto evt = v2::CreateNumericOutcome(fbb, outcome).Union();
+        const auto idx = fbb.CreateString(index).Union();
+        auto fb = v2::CreateOutcomeEvent(fbb, v2::OutcomeValue_numeric, evt, v2::IndexValue_literal, idx);
         fbb.Finish(fb);
         return fbb.Release();
       }
 
       static generic_event::payload_buffer_t string_event(int index, const char* outcome) {
         flatbuffers::FlatBufferBuilder fbb;
-        const auto evt = v2::CreateStringEventIndexedDirect(fbb, outcome, index).Union();
-        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_StringEventIndexed, evt);
+        const auto evt = fbb.CreateString(outcome).Union();
+        const auto idx = v2::CreateNumericIndex(fbb, index).Union();
+        auto fb = v2::CreateOutcomeEvent(fbb, v2::OutcomeValue_literal, evt, v2::IndexValue_numeric, idx);
+        fbb.Finish(fb);
+        return fbb.Release();
+      }
+
+      static generic_event::payload_buffer_t string_event(const char* index, const char* outcome) {
```

Comment:
potential ambiguity between int / char* (as you pointed out before)

Reply:
I looked at this closely and this doesn't happen. Int won't be implicitly cast to const char*